### ISSUE TITLE
Full butchery ignores nearby "wooden table" item

### ISSUE THREAD

**Issue by l29ah** [CleverRaven/Cataclysm-DDA#28511]
**To Reproduce**  
1. Prepare a knife, long rope and a nearby tree.
2. `&`Craft wooden table (fabrication 1, 20 minutes).
3. Try to butcher something on it.
4. "For a corpse this big you need a table nearby or something else with a flat surface to perform a full butchery."

**Screenshots**  
![](https://my.mixtape.moe/qlcimu.png)

**Versions and configuration(please complete the following information):**
0.C-37429-g2365f4b02b-dirty

**Comment by kevingranade** [comment]
Most likely it's just missing a flag.

**Comment by l29ah** [comment]
```
    bool has_table_nearby = false;
    for( const tripoint &pt : g->m.points_in_radius( u.pos(), 2 ) ) {
        if( g->m.has_flag_furn( "FLAT_SURF", pt ) || g->m.has_flag( "FLAT_SURF", pt ) ||
            ( g->m.veh_at( pt ) && g->m.veh_at( pt )->vehicle().has_part( "KITCHEN" ) ) ) {
            has_table_nearby = true;
        }
    }
```
Seems like it won't account for neither a table item nor a table installed on a vehicle.

**Comment by kevingranade** [comment]
It's intended that it doesn't look for a table item.
Looks like you can add something like ``g->m.veh_at( pt )->has_feature( "FLAT_SURF" )`` to the above code and add "FLAT_SURF" to
https://github.com/CleverRaven/Cataclysm-DDA/blob/b42b941b9b25d25128d2b087f35af5ee1bb511a8/data/json/vehicle_parts.json#L978-L998
and it might do what you want.

**Comment by ZhilkinSerg** [comment]
Fixed in #28531 (you need to place crafted table).

**Comment by kevingranade** [comment]
It would still be good to check for a usable vehicle table.
